contrib/database/sql/conn.go

```diff
@@ -167,6 +211,48 @@ func WithSpanTags(ctx context.Context, tags map[string]string) context.Context {
 	return context.WithValue(ctx, spanTagsKey, tags)
 }
 
+// tryStartTrace will create a span using the given arguments, but will act as a no-op when err is driver.ErrSkip.
+func (tp *traceParams) tryStartTrace(ctx context.Context, qtype queryType, query string, startTime time.Time, sqlCommentCarrier *tracer.SQLCommentCarrier) (span tracer.Span) {
+	if _, exists := tracer.SpanFromContext(ctx); tp.cfg.childSpansOnly && !exists {
+		return nil
+	}
+	name := fmt.Sprintf("%s.query", tp.driverName)
+	opts := []ddtrace.StartSpanOption{
+		tracer.ServiceName(tp.cfg.serviceName),
+		tracer.SpanType(ext.SpanTypeSQL),
+		tracer.StartTime(startTime),
+	}
+	if !math.IsNaN(tp.cfg.analyticsRate) {
+		opts = append(opts, tracer.Tag(ext.EventSampleRate, tp.cfg.analyticsRate))
+	}
+	span, _ = tracer.StartSpanFromContext(ctx, name, opts...)
+	resource := string(qtype)
+	if query != "" {
+		resource = query
+	}
+	span.SetTag("sql.query_type", string(qtype))
+	span.SetTag(ext.ResourceName, resource)
+	for k, v := range tp.meta {
+		span.SetTag(k, v)
+	}
+	if meta, ok := ctx.Value(spanTagsKey).(map[string]string); ok {
+		for k, v := range meta {
+			span.SetTag(k, v)
+		}
+	}
+
+	if sqlCommentCarrier != nil && tp.cfg.commentInjectionMode != commentInjectionDisabled {
+		injectionOpts := injectionOptionsForMode(tp.cfg.commentInjectionMode, sqlCommentCarrier.DiscardDynamicTags)
+		err := tracer.InjectWithOptions(span.Context(), sqlCommentCarrier, injectionOpts...)
+		if err != nil {
+			// this should never happen
+			fmt.Fprintf(os.Stderr, "contrib/database/sql: failed to inject query comments: %v\n", err)
```

Comment:
You probably want to use the internal `log` package here instead of just calling `Fprintf` directly